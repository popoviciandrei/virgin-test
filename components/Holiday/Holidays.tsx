import React from 'react';
import { HolidayTile } from '@components/Holiday/';

import { FetchHolidaysResponse as Props } from '@lib/api/types';

const Holidays = ({ holidays }: Props): JSX.Element => {
  if (!holidays?.length) {
    return <h2>No holidays found!</h2>;
  }

  return (
    <>
      <h1>{holidays.length} holidays found</h1>
      <ul>
        {holidays.map((holiday, index) => (
          <HolidayTile holiday={holiday} key={index} />
        ))}
      </ul>
    </>
  );
};

export default Holidays;
