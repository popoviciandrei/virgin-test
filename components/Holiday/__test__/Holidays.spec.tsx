import React from 'react';
import { render } from '@testing-library/react';
import Holidays from '../Holidays';

jest.mock('@components/Holiday', () => ({
  __esModule: true,
  HolidayTile: (props): JSX.Element => <div data-testid="HolidayTile" {...props} />,
}));

describe('Holidays', () => {
  it('should render the Holidays', () => {
    const { container } = render(<Holidays holidays={['holiday 1', 'holiday 2']} />);
    expect(container).toMatchSnapshot();
  });
  it('should render no holidays', () => {
    const { container } = render(<Holidays />);
    expect(container).toMatchSnapshot();
  });
});
