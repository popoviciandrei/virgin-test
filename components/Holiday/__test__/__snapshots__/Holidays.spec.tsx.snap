// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Holidays should render no holidays 1`] = `
<div>
  <h2>
    No holidays found!
  </h2>
</div>
`;

exports[`Holidays should render the Holidays 1`] = `
<div>
  <h1>
    2
     holidays found
  </h1>
  <ul>
    <div
      data-testid="HolidayTile"
      holiday="holiday 1"
    />
    <div
      data-testid="HolidayTile"
      holiday="holiday 2"
    />
  </ul>
</div>
`;
